lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4468,9 +4468,11 @@ std::string SPIRVToLLVM::getOCLGenericCastToPtrName(SPIRVInstruction *BI) {
 
 llvm::GlobalValue::LinkageTypes
 SPIRVToLLVM::transLinkageType(const SPIRVValue *V) {
-  if (V->getLinkageType() == internal::LinkageTypeInternal) {
+  SPIRVLinkageTypeKind LT = V->getLinkageType();
+  if (LT == internal::LinkageTypeInternal)
```

Comment:
Can we check this linkage type with a `case` in the `switch(LT)` statement below?

Reply:
Checking this in the switch statement gives a warning because `LinkageTypeInternal` is not a part of `SPIRVLinkageTypeKind` enum. It is defined in spirv_internal, not in spirv.hpp like other linkage types. I see this:
```
SPIRVReader.cpp:4475:3: warning: case value ‘2147483646’ not in enumerated type ‘SPIRV::SPIRVLinkageTypeKind {aka spv::LinkageType}’ [-Wswitch]
   case internal::LinkageTypeInternal:
```

Reply:
As w/a I can suggest to cast LT to `int` and switch over it. No warnings shall be emitted.

Reply:
Okay, done.